tests/fsharp/Compiler/Language/ComputedCollectionLoweringTests.fs

```diff
@@ -0,0 +1,790 @@
+                  {
+                    .field public int32 start
+                    .method assembly specialname rtspecialname 
+                            instance void  .ctor(int32 start) cil managed
+                    {
+                      .custom instance void [runtime]System.Runtime.CompilerServices.CompilerGeneratedAttribute::.ctor() = ( 01 00 00 00 ) 
+                      .custom instance void [runtime]System.Diagnostics.DebuggerNonUserCodeAttribute::.ctor() = ( 01 00 00 00 ) 
+                      
+                      .maxstack  8
+                      IL_0000:  ldarg.0
+                      IL_0001:  call       instance void class [FSharp.Core]Microsoft.FSharp.Core.FSharpFunc`2<int32,int32>::.ctor()
+                      IL_0006:  ldarg.0
+                      IL_0007:  ldarg.1
+                      IL_0008:  stfld      int32 Test/test@1::start
+                      IL_000d:  ret
+                    } 
+                
+                    .method public strict virtual instance int32 
+                            Invoke(int32 i) cil managed
+                    {
+                      
+                      .maxstack  8
+                      IL_0000:  ldarg.0
+                      IL_0001:  ldfld      int32 Test/test@1::start
+                      IL_0006:  ldarg.1
+                      IL_0007:  add
+                      IL_0008:  ret
+                    } 
+                
+                  } 
+                
+                  .method public static int32[]  test(int32 start,
+                                                      int32 finish) cil managed
+                  {
+                    .custom instance void [FSharp.Core]Microsoft.FSharp.Core.CompilationArgumentCountsAttribute::.ctor(int32[]) = ( 01 00 02 00 00 00 01 00 00 00 01 00 00 00 00 00 ) 
+                    
+                    .maxstack  8
+                    IL_0000:  ldarg.0
+                    IL_0001:  ldarg.1
+                    IL_0002:  bgt.s      IL_0016
+                
+                    IL_0004:  ldarg.1
+                    IL_0005:  ldarg.0
+                    IL_0006:  sub
+                    IL_0007:  ldc.i4.1
+                    IL_0008:  add
+                    IL_0009:  ldarg.0
+                    IL_000a:  newobj     instance void Test/test@1::.ctor(int32)
+                    IL_000f:  call       !!0[] [FSharp.Core]Microsoft.FSharp.Collections.ArrayModule::Initialize<int32>(int32,
+                                                                                                                        class [FSharp.Core]Microsoft.FSharp.Core.FSharpFunc`2<int32,!!0>)
+                    IL_0014:  ret
+                
+                    IL_0015:  ldnull
+                    IL_0016:  call       !!0[] [runtime]System.Array::Empty<int32>()
+                    IL_001b:  ldnull
+                    IL_001c:  ret
```

Comment:
I don't know where these `ldnull`s are coming from.

Reply:
Those are for sure a regression, I do not see them if handrolling equivalent code.
Does it go away if you remove the `  start > finish ` special case?

Reply:
0136a96